pkg/rpc/rpc.go

```diff
@@ -0,0 +1,61 @@
+package rpc
+
+import (
+	"fmt"
+
+	rpcclient "github.com/cometbft/cometbft/rpc/client/http"
+	"github.com/cosmos/cosmos-sdk/client"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"google.golang.org/grpc"
+
+	authoritytypes "github.com/zeta-chain/zetacore/x/authority/types"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+	lightclienttypes "github.com/zeta-chain/zetacore/x/lightclient/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// Clients contains RPC client interfaces to interact with zetacored
+type Clients struct {
+	AuthorityClient authoritytypes.QueryClient
```

Comment:
Can we use embedding instead? (if these clients don't share same the method names)
```suggestion
	authoritytypes.QueryClient
```

This will simplify duck typing in cases when we want to pass a client with a subset of methods

Reply:
Interesting let me try it.

Reply:
Since they're all named `QueryClient` it seems you can't embed them. I guess I could type alias them but I'm not sure that would be great.

I think it might actually be a bit better to use the name so you know what types you need to import anyway.

Reply:
Okay, makes sense. Then let's keep it as is but remove `*Client` suffix.
```go
type Clients struct {
	Authority authoritytypes.QueryClient
        // ...
}
```

Reply:
Yep already done :)